communication/src/main/java/org/jboss/da/communcation/pnc/PNC.java

```diff
@@ -1,34 +1,68 @@
 package org.jboss.da.communcation.pnc;
 
-import org.jboss.da.communcation.pnc.entity.BuildConfiguration;
-import org.jboss.da.communcation.pnc.entity.BuildConfigurationSet;
-import org.jboss.da.communcation.pnc.entity.Product;
-import org.jboss.da.communcation.pnc.entity.Project;
+import org.jboss.da.common.json.DAConfig;
+import org.jboss.da.common.util.Configuration;
+import org.jboss.da.common.util.ConfigurationParseException;
+import org.jboss.da.communcation.pnc.authentication.PNCAuthentication;
+import org.jboss.da.communcation.pnc.model.BuildConfiguration;
+import org.jboss.da.communcation.pnc.model.BuildConfigurationSet;
+import org.jboss.da.communcation.pnc.model.Product;
+import org.jboss.da.communcation.pnc.model.Project;
+import org.jboss.resteasy.client.ClientRequest;
+import org.jboss.resteasy.client.ClientResponse;
 
-import javax.ws.rs.*;
 import javax.ws.rs.core.MediaType;
+import java.util.Arrays;
 import java.util.List;
 
-@Path("pnc-rest/rest")
-public interface PNC {
+public class PNC {
 
-    @GET
-    @Path("build-configurations")
-    @Produces(MediaType.APPLICATION_JSON)
-    List<BuildConfiguration> getBuildConfigurations();
+    private Configuration config;
+    private DAConfig conf;
+    private String pncServer;
+    private String token;
 
-    @GET
-    @Path("build-configuration-sets")
-    @Produces(MediaType.APPLICATION_JSON)
-    List<BuildConfigurationSet> getBuildConfigurationSets();
+    public PNC() throws ConfigurationParseException {
+        Configuration config = new Configuration();
+        DAConfig conf = config.getConfig();
+        pncServer = config.getConfig().getPncServer();
 
-    @GET
-    @Path("products")
-    @Produces(MediaType.APPLICATION_JSON)
-    List<Product> getProducts();
+        PNCAuthentication pncAuthenticate = new PNCAuthentication();
+        token = pncAuthenticate.authenticate();
```

Comment:
Is there an option to check if we are authenticated? But for now it could try to authenticate each time, but create an issue for it please.